File: Fundamentals_of_Software_Design/1_Basic_skills/Числа Фибоначчи/main.c
#include <stdio.h>

long long fibonacci(int n) {
    if (n <= 0) return 0;
    if (n == 1) return 1;

    long long a = 0, b = 1, temp;
    for (int i = 2; i <= n; i++) {
        temp = a + b;
        a = b;
        b = temp;
    }
    return b;
}

int main() {
    int n;
    printf("Введите n: ");
    scanf("%d", &n);
    printf("F(%d) = %lld\n", n, fibonacci(n));
    return 0;
}

File: Fundamentals_of_Software_Design/1_Basic_skills/Числа Фибоначчи/main.py
def fibonacci(n):
    if n <= 0:
        return 0
    elif n == 1:
        return 1

    a, b = 0, 1
    for _ in range(2, n + 1):
        a, b = b, a + b
    return b

n = int(input("Введите n: "))
print(f"F({n}) = {fibonacci(n)}")
